tests/test_acl.rb

```diff
@@ -0,0 +1,161 @@
+    rtr = Acl.new(acl_name, afi)
+    assert(Acl.acls[afi].key?(acl_name),
+           "ACL #{afi} #{acl_name} is not configured")
+
+    @default_show_command = "show runn | i '#{afi} access-list #{acl_name}'"
+    assert_show_match(pattern: /^#{afi} access-list #{acl_name}$/,
+                      msg:     "failed to create acl #{acl_name}")
+    rtr.destroy
+    refute_show_match(pattern: /^#{afi} access-list #{acl_name}$/,
+                      msg:     "failed to destroy acl #{acl_name}")
+  end
+
+  def test_create_acl
+    %w(ip ipv6).each do |afi|
+      acl_name = afi[/ipv6/] ? @acl_name_v6 : @acl_name_v4
+      create_acl(afi, acl_name)
+    end
+  end
+
+  def stats_enable(afi, acl_name)
+    rtr = Acl.new(acl_name, afi)
+
+    @default_show_command = "show runn | sec '#{afi} access-list #{acl_name}'"
+    assert_show_match(pattern: /^#{afi} access-list #{acl_name}$/,
+                      msg:     "failed to create acl #{acl_name}")
+
+    # setter function
+    rtr.stats_per_entry = true
+    assert_show_match(pattern: /statistics per-entry/,
+                      msg:     'failed to enable stats')
+
+    # getter function
+    val = rtr.stats_per_entry
+    assert(val, 'value is not true')
+
+    # default getter function
+    val = rtr.default_stats_per_entry
+    refute(val, 'value is not false')
+
+    rtr.destroy
+    refute_show_match(pattern: /^ip access-list #{acl_name}$/,
+                      msg:     "failed to destroy acl #{acl_name}")
+  end
+
+  def test_stats_enable
+    %w(ip ipv6).each do |afi|
+      acl_name = afi[/ipv6/] ? @acl_name_v6 : @acl_name_v4
+      stats_enable(afi, acl_name)
+    end
+  end
+
+  def set_fragments(rtr, afi, acl_name, option)
+    @default_show_command = "show runn | sec '#{afi} access-list #{acl_name}'"
+
+    # setter function
+    rtr.fragments = option
+    assert_show_match(pattern: /fragments #{option}/,
+                      msg:     'failed to set fragments #{option} ' + option)
+
+    # getter function
+    val = rtr.fragments
+    assert_equal(val, 'fragments ' + option, 'value not correct')
+  end
+
+  def unset_fragments(rtr, afi, acl_name)
+    @default_show_command = "show runn | sec '#{afi} access-list #{acl_name}'"
+
+    # setter function
+    rtr.fragments = nil
+    refute_show_match(pattern: /fragments #{@permit}|fragments #{@deny}/,
+                      msg:     'failed to unset set fragments')
+
+    # getter function
+    val = rtr.fragments
+    assert_equal(val, 'nil', 'value not correct')
+  end
+
+  def fragments(afi, acl_name)
+    rtr = Acl.new(acl_name, afi)
+
+    @default_show_command = "show runn | sec '#{afi} access-list #{acl_name}'"
+    assert_show_match(pattern: /^#{afi} access-list #{acl_name}$/,
+                      msg:     "failed to create acl #{acl_name}")
+
+    # set 'no frgaments ...'
+    unset_fragments(rtr, afi, acl_name)
+
+    # set 'fragments permit-all' from nothing set
+    set_fragments(rtr, afi, acl_name, @permit)
```

Comment:
nit: These `@permit` `@deny` class vars are okay but you're basically using them as constants. And they don't save that much typing so I would just use `permit-all` and `deny-all` instead. Just a style preference - keep what you have if you like that better.